### PR TITLE
[13.0][IMP] stock_secondary_unit: Use secondary units directly in pickings

### PR DESCRIPTION
**Pull request by sergio-teruel**
[IMP] stock_secondary_unit: Use product_secondary_unit mixin model from product-attribute repository.

cc @Tecnativa TT27363
ping @carlosdauden @chienandalu

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
LGTM :+1:

File: stock_secondary_unit/models/stock_move.py [hunk 1]
```diff
@@ -4,31 +4,49 @@
...
 
     def _merge_moves_fields(self):
         res = super()._merge_moves_fields()
-        res["secondary_uom_qty"] = self[-1:].secondary_uom_qty
+        res["secondary_uom_qty"] = sum(self.mapped("secondary_uom_qty"))
         return res
 
+    @api.model
+    def _prepare_merge_moves_distinct_fields(self):
+        # Not merge moves with distinct secondary units
```
Comment (pedrobaeza) [inline stock_secondary_unit/models/stock_move.py]: You should put this as a docstring:
```suggestion
        """Don't merge moves with distinct secondary units."""
```

File: stock_secondary_unit/models/stock_move.py [hunk 2]
```diff
@@ -4,31 +4,49 @@
...
+        return distinct_fields
+
 
 class StockMoveLine(models.Model):
-    _inherit = ["stock.move.line", "stock.secondary.unit.mixin"]
+    _inherit = ["stock.move.line", "product.secondary.unit.mixin"]
     _name = "stock.move.line"
+    _secondary_unit_fields = {"qty_field": "qty_done", "uom_field": "product_uom_id"}
+
+    secondary_uom_qty = fields.Float()
```
Comment (pedrobaeza) [inline stock_secondary_unit/models/stock_move.py]: Why do you need to declare these fields?

File: stock_secondary_unit/models/stock_move.py [hunk 3]
```diff
@@ -4,31 +4,49 @@
...
+    def _prepare_merge_moves_distinct_fields(self):
+        # Not merge moves with distinct secondary units
+        distinct_fields = super()._prepare_merge_moves_distinct_fields()
+        distinct_fields += ["secondary_uom_id"]
+        return distinct_fields
+
 
 class StockMoveLine(models.Model):
-    _inherit = ["stock.move.line", "stock.secondary.unit.mixin"]
+    _inherit = ["stock.move.line", "product.secondary.unit.mixin"]
```
Comment (pedrobaeza) [inline stock_secondary_unit/models/stock_move.py:41]: UoM is not editable in these lines so no need of onchange?

File: stock_secondary_unit/models/stock_move.py (hunk 2 above)
Comment (sergio-teruel) [reply stock_secondary_unit/models/stock_move.py]: Good question.... yes... yes....
Due to compute order fields....

File: stock_secondary_unit/models/stock_move.py (hunk 3 above)
Comment (sergio-teruel) [reply stock_secondary_unit/models/stock_move.py:41]: Yes... but you can create a new stock move lines directly...

**Comment by sergio-teruel** [comment]
Changes done

File: stock_secondary_unit/models/stock_move.py (hunk 2 above)
Comment (pedrobaeza) [reply stock_secondary_unit/models/stock_move.py]: I'm not understanding you, but any explanation should be put here as comment for later references.

File: stock_secondary_unit/models/stock_move.py (hunk 3 above)
Comment (pedrobaeza) [reply stock_secondary_unit/models/stock_move.py:41]: I also don't understand you. Is it needed or not? If so, I don't see it in your current code.

File: stock_secondary_unit/models/stock_move.py (hunk 2 above)
Comment (sergio-teruel) [reply stock_secondary_unit/models/stock_move.py]: Solved with this PR https://github.com/OCA/product-attribute/pull/773

**Pull request merged**